mesonbuild/mesonlib.py

```diff
@@ -1115,7 +1115,7 @@ def unholder(item: T.List[_T]) -> T.List[_T]: ...
 @T.overload
 def unholder(item: T.List[T.Union[_T, 'ObjectHolder[_T]']]) -> T.List[_T]: ...
 
-def unholder(item):
+def unholder(item):  # type: ignore  # TODO fix overload (somehow)
```

Comment:
I think the right thing to do to fix a lot of these to is to insist that all flattening happen in the interpreter, that way all other functions can safely assume they're being passed flat data structures. That's another refactor for another time thought

Reply:
Then I would leave this as `type: ignore` for now.